File: app_telefono/app/src/main/java/com/canut/sprint_0/MainActivity.java
package com.canut.sprint_0;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.bluetooth.BluetoothAdapter;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private int REQUEST_ENABLE_BT = 1;
    static public final int REQUEST_LOCATION = 1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);





        BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();

        if (!bluetoothAdapter.isEnabled()) {
            Intent enableBtIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
            startActivityForResult(enableBtIntent, REQUEST_ENABLE_BT);
        }






        final ReceptorBLE receptorBLE = new ReceptorBLE(this, bluetoothAdapter);





        Button btn_start = findViewById(R.id.btn_start);
        btn_start.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Log.e("HOLA", "HOLA DIEGO");
                receptorBLE.startScanning();
            }
        });

        Button btn_stop = findViewById(R.id.btn_stop);
        btn_stop.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Log.e("ADIOS", "ADIOS DIEGO");
                receptorBLE.stopScanning();
            }
        });


    }

}

File: app_telefono/app/src/main/java/com/canut/app_sprint0/MainActivity.kt
package com.canut.app_sprint0

// ---------------------------------------------------
//  MainActivity
//  Encargado de recoger los datos enviados por el iBeacon y enviarlos al servidor.
//  Carlos Mª Canut Dominguez
//  13-10-2019
// ---------------------------------------------------


import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import com.canut.app_sprint0.LogicaTelefono
import com.canut.app_sprint0.PeticionarioREST

class MainActivity : AppCompatActivity() {

    // ---------------------------------------------------
    //  ->
    //  onCreate()
    //  ->
    // ---------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val logica = LogicaTelefono()

        logica.start(btn_start)
        logica.stop(btn_stop)



    }
}
